fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/config/SecurityConfig.java

```diff
@@ -165,7 +166,9 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 .addFilterBefore(tenantAwareBasicAuthenticationFilter(), SecurityContextHolderFilter.class) //
                 .addFilterAfter(requestResponseFilter(), ExceptionTranslationFilter.class) //
                 .addFilterAfter(correlationHeaderFilter(), RequestResponseFilter.class) //
-                .addFilterAfter(fineractInstanceModeApiFilter(), CorrelationHeaderFilter.class); //
+                .addFilterAfter(fineractInstanceModeApiFilter(), CorrelationHeaderFilter.class) //
+                .addFilterAfter(fineractInstanceModeApiFilter(), CorrelationHeaderFilter.class) //
+                .addFilterAfter(geolocationHeaderFilter(), RequestResponseFilter.class); //
```

Comment:
You can decide to not register this filter if the functionality is disabled... See example: `loanCOBFilterHelper`